### PR TITLE
GH-2268 Update the limit param in graphql api's

### LINKED ISSUE #2268: Update the limit param in graphql api's
Currently, the `limit` param allows you to specify a limit when retrieving items back from an API call. This is a numeric value that can be passed in but in some cases, we may want all results. To address this, postress allows us to pass in `ALL` instead of a numeric value, which is effectively the same as not passing a limit at all. Update all of the graphql API endpoints such that if `None` is passed in it will pass `ALL` to the LIMIT clause in the query. This will allow us to get all entries without modifying the existing api.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #2268.

Currently, the limit param allows you to specify a limit when retrieving items back from an API call. This is a numeric value that can be passed in but in some cases, we may want all results. To address this, postress allows us to pass in ALL instead of a numeric value, which is effectively the same as not passing a limit at all. Update all of the graphql API endpoints such that if None is passed in it will pass ALL to the LIMIT clause in the query. This will allow us to get all entries without modifying the existing api.

Signed-off-by: johnsonw <wjohnson@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2269)

### CODE REVIEW

File: iml-api/src/graphql.rs [hunk 1]
```diff
@@ -173,7 +173,7 @@ impl QueryRoot {
                 OFFSET $2 LIMIT $3"#,
             dir.deref(),
             offset.unwrap_or(0) as i64,
-            limit.unwrap_or(20) as i64
+            limit.map(|x| x.to_string()).unwrap_or("ALL".to_string()) as String
```
Comment (ip1981) [inline iml-api/src/graphql.rs:176]:
```suggestion
            limit.map(|x| x.to_string()).unwrap_or_else(|| "ALL".to_string()) as String
```

Comment (johnsonw) [reply iml-api/src/graphql.rs:176]: sure, makes sense. I'll update.

**Review by jgrund — CHANGES_REQUESTED** [review]
Comments need updating as well, as they still say they default to 20.

File: iml-api/src/graphql.rs [hunk 2]
```diff
@@ -173,7 +173,9 @@ impl QueryRoot {
                 OFFSET $2 LIMIT $3"#,
             dir.deref(),
             offset.unwrap_or(0) as i64,
-            limit.unwrap_or(20) as i64
+            limit
+                .map(|x| x.to_string())
+                .unwrap_or_else(|| "ALL".to_string()) as String
```
Comment (ip1981) [inline iml-api/src/graphql.rs:178]: `as String` really needed?

Comment (johnsonw) [reply iml-api/src/graphql.rs:178]: It required the cast.

File: iml-api/src/graphql.rs [hunk 3]
```diff
@@ -136,7 +136,7 @@ pub(crate) struct QueryRoot;
 #[juniper::graphql_object(Context = Context)]
 impl QueryRoot {
     #[graphql(arguments(
-        limit(description = "paging limit, defaults to 20"),
+        limit(description = "paging limit, defaults to All"),
```
Comment (jgrund) [inline iml-api/src/graphql.rs:139]: Maybe: 

```suggestion
        limit(description = "optional paging limit, defaults to all rows"),
```

**Review by jgrund — APPROVED** [review]

**Review by ip1981 — APPROVED** [review]

**Pull request merged**